components/supervisor/pkg/ports/ports.go

```diff
@@ -411,6 +411,11 @@ func (pm *Manager) nextState(ctx context.Context) map[uint32]*managedPort {
 
 		var public bool
 		config, kind, exists := pm.configs.Get(mp.LocalhostPort)
+		if exists {
```

Comment:
I wonder why it is necessary here, configurations should be handled why earlier above, this section about served state

Reply:
ah, it is for ranges?

Reply:
visibility (port private and public) is handled here https://github.com/gitpod-io/gitpod/blob/3013055d321c0eabff7216541f847d0df53c2c8f/components/supervisor/pkg/ports/ports.go#L430

Reply:
Do we need to call `mp.OnExposed = getOnExposedAction(config, port)`?

Reply:
> Do we need to call mp.OnExposed = getOnExposedAction(config, port)? or it is handled already somewhere too?
added

I tested with `onOpen: open-browser` with ranges before and it works fine, but I don't know where is code. Add to confirm it

Reply:
thank you, great 👍

Reply:
maybe a commit in if will be nice, that we are handling configuration for port ranges here